mojave/code/modules/mob/living/carbon/human/human_armor.dm

```diff
@@ -74,6 +74,22 @@
 	return protection
 
 /mob/living/carbon/human/damage_armor(damage = 0, damage_flag = MELEE, damage_type = BRUTE, sharpness = NONE, def_zone = BODY_ZONE_CHEST)
+	//We need to convert attack flags into actually useful subarmor variables
+	var/static/list/conversion_table = list(MELEE, BULLET)
+	if(damage_flag in conversion_table)
+		damage_flag = CRUSHING
+		if(sharpness & SHARP_IMPALING)
+			damage_flag = IMPALING
+		else if(sharpness & SHARP_POINTY)
+			damage_flag = PIERCING
+		else if(sharpness & SHARP_EDGED)
+			damage_flag = CUTTING
+
+	if(wear_suit && istype(wear_suit, /obj/item/clothing/suit/space/hardsuit/ms13/power_armor))
+		if((def_zone == BODY_ZONE_HEAD || def_zone == BODY_ZONE_PRECISE_EYES || def_zone == BODY_ZONE_PRECISE_MOUTH) && !wear_suit:helmettype)
+			return -damage
+		return wear_suit.get_integrity() - wear_suit.take_damage(damage, damage_type, damage_flag, 0, def_zone = def_zone)
```

Comment:
Why return `-damage` when most calls to `damage_armor` are set to negative as well?  Is this just to account for line 91 where the return value could be positive?  Would it just be easier to return `damage` without setting it to negative and then for line 91 just adding a conditional to return a value of 0 if the damage doesn't get past the armor integrity?